src/main/webapp/app/scores/exercise-scores.component.ts

```diff
@@ -55,12 +61,19 @@ export class ExerciseScoresComponent implements OnInit, OnDestroy {
 
     ngOnInit() {
         this.paramSub = this.route.params.subscribe(params => {
+            this.isLoading = true;
             this.courseService.find(params['courseId']).subscribe((res: HttpResponse<Course>) => {
                 this.course = res.body!;
             });
             this.exerciseService.find(params['exerciseId']).subscribe((res: HttpResponse<Exercise>) => {
                 this.exercise = res.body!;
-                this.getResults();
+                // We need to preload the pending submissions here, otherwise every updating-result would trigger a single REST call.
+                const additionalLoadingOperation =
```

Comment:
I would give this a more concrete name, like `loadAndCacheAllSubmissionStatesOperation`

Reply:
Looking at this again, it was not very nice. I moved the logic to a function with a better name that you proposed.